lib/reqIdGenFactory.js

```diff
@@ -9,6 +9,7 @@ module.exports = function () {
   const maxInt = 2147483647
   let nextReqId = 0
   return function genReqId (req) {
-    return (nextReqId = (nextReqId + 1) & maxInt)
+    nextReqId = (nextReqId + 1) & maxInt
+    return `req-${nextReqId.toString(36)}`
```

Comment:
This output would be `req-1` if i'm not wrong. Was it wanted?

Reply:
yes

Reply:
@mcollina @Eomm I apologize for not including tests in the first place; maybe it would be easier to distinguish the intent of the change if tests are provided.

Do you think the project would benefit from tests for this case?
If so I can open a new PR with tests case(s) covering this change.

Reply:
I asked because I read the issue and I understood that the outcome would be a string like the "base64", so it was not clear to me

In this case I'm not sure what should do the to String(32) function since the variable is a number

Reply:
After joining a discussion on the issue it was concluded that a prefix of `req-` should be added alongside of converting the number to a string with a base of 36.

`toString` method on the `Number.prototype` returns a string representing the specified number (on which the method was called) using the specified `radix` (base).
This will return a string representing the number from the `10` base (in the example you provided `1`) to a string with a base of `36` (which in the same example will result in the value `1`, like in the base 10).

However, the difference comes for the numbers greater than 9.
i.e. for the number value `25752` (base 10) the output would be `64a2` (base 36).

To better understand this please do refer to the following (taken from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Bad_radix)):

> The optional radix parameter of the Number.prototype.toString() or the BigInt.prototype.toString() method was specified. Its value must be an integer (a number) between 2 and 36, specifying the base of the number system to be used for representing numeric values.

> The most common radixes:
2 for binary numbers,
8 for octal numbers,
10 for decimal numbers,
16 for hexadecimal numbers.

I apologize if I completely misunderstood your question and just stated the obvious, but just wanted to give a little background of the change and why was the `toString(36)` used in this context.

Also, these comments were the ones on which the implementation was based upon:
https://github.com/fastify/fastify/issues/2736#issuecomment-745247740
https://github.com/fastify/fastify/issues/2736#issuecomment-777382265

Reply:
Thanks for the explanation, I was not aware of the `toString` supports the `base` parameter too!